File: Circle/Circle.ts
import { Figure } from '../Figure/Figure'

/**
 * Класс Circle экспортируем и реализует интерфейс
 * в этом классе находится логика фигуры
 */
export class Circle implements Figure {
     type = 'circle';// Тип фигуры - круг
     private Radiuse: number;// Радиус круга

     constructor(radius: number) { 
          this.Radiuse = radius;
     }
     // Геттер для получения радиуса круга
     get radius() {
          return this.Radiuse;
     }
    /**
      * Метод для вычисления площади круга
      * @returns Площадь круга по формуле Math.PI * radius^2
      */
    public getArea(): number {
     return Math.PI * this.radius * this.radius;
   }
      /**
      * Метод для вычисления периметра круга
      * @returns Периметр круга по формуле 2 * Math.PI * radius
      */
     getPerimeter() {
          // Возвращаем периметр круга по формуле
          return 2 * Math.PI * this.Radiuse;
     }
}

File: Canculate_Figur/Canculate_Figur/Circle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Canculate_Figur
{
    class Circle:Figure
    {
        private double radius;

        public Circle(double radius)
        {
            this.radius = radius;
        }

        //Вычисления площади
        public double GetArea()
        {
            return Math.PI * radius * radius;
        }

        //Вычисления периметра 
        public double GetPerimeter()
        {
            return 2 * Math.PI * radius;
        }

        //Получения имени фигуры
        public string GetName()
        {
            return "Круг";
        }
    }
}
